src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -3179,23 +3179,38 @@ public static Vector128<ulong> ShiftRightLogical(Vector128<ulong> vector, int sh
         /// <param name="indices">The per-element indices used to select a value from <paramref name="vector" />.</param>
         /// <returns>A new vector containing the values from <paramref name="vector" /> selected by the given <paramref name="indices" />.</returns>
         [Intrinsic]
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static Vector128<byte> Shuffle(Vector128<byte> vector, Vector128<byte> indices)
         {
-            Unsafe.SkipInit(out Vector128<byte> result);
-
-            for (int index = 0; index < Vector128<byte>.Count; index++)
+            if (Ssse3.IsSupported)
+            {
+                Debug.Assert(BitConverter.IsLittleEndian);
```

Comment:
Afair, the current impl of this intrinsic does not directly map into Ssse3.Shuffle/AdvSimd.Arm64.VectorTableLookup so this change might not be valid cc @tannergooding

Reply:
Well spotted. The behaviour would differ on x86 when 0x80 > index > 0xF, it will be mapped to `index & 0x0F` position. I will check this further.

Reply:
`Vector128/256.Shuffle` behaves like it does on Arm64/WASM which is that any out of range index is treated as "zero this element".
`Ssse3.Shuffle` on the other hand treats it as "highest bit set means zero this element; otherwise mask off to the lowest 4 bits"

For comparison, `mask: 0x98` means:
```csharp
Vector128.Shuffle: Zero destination element
Ssse3.Shuffle: Zero destination element
```

while, `mask: 0x18` means:
```csharp
Vector128.Shuffle: Zero destination element
Ssse3.Shuffle: Set destination element to source.GetElement(8)
```

`Vector256.Shuffle` likewise differs and treats it as a contiguous 256-bit vector rather than 2x 128-bit vectors.

Reply:
Switched to use `Vector128.Shuffle()` instead of architecture specific shuffles that emits `vpshufd` on x86 and `tbl` for AArch64.